lib/game.rb

```diff
@@ -0,0 +1,122 @@
+class Game
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 4, 8],
+    [2, 4, 6],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8]
+    ]
+
+  def initialize(player_1 = Human.new("X"), player_2 = Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  #-----returns the next player-----#
+  def current_player
+    board.turn_count.even? ? player_1 : player_2
+  end
+
+  def over?
+    won? || draw?
+  end
+
+  #-----iterate over win_combo array and compare with current board pos to see if all "X"s or all "O"s-----#
+  def won?
+    WIN_COMBINATIONS.detect do |combo|
+      board.position(combo[0]+1) == board.position(combo[1]+1) &&
+      board.position(combo[1]+1) == board.position(combo[2]+1) &&
+      board.taken?(combo[0]+1)
+    end
+  end
+
+  def draw?
+    board.full? && !won?
+  end
+
+  #-----use results from won? to return player token-----#
+  def winner
+    board.position(won?[0]+1) if won?
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.